Mathlib/MeasureTheory/Function/UnifTight.lean

```diff
@@ -0,0 +1,44 @@
+/-
+Copyright (c) 2023 Igor Khavkine. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Igor Khavkine
+-/
+import Mathlib.MeasureTheory.Function.LpSeminorm.Basic
+
+/-!
+# Uniform tightness
+
+This file contains the definitions for uniform tightness for a family of Lp functions.
+It is used as a hypothesis to the version of Vitali's convergence theorem for Lp spaces
+that works also for spaces of infinite measure.
+
+## Tags
+
+uniform integrable, uniformly tight, Vitali convergence theorem
+-/
+
+
+namespace MeasureTheory
+
+open Classical Set Filter Topology MeasureTheory NNReal ENNReal
+
+variable {α β ι : Type*} {m : MeasurableSpace α} {μ : Measure α} [NormedAddCommGroup β]
+
+
+
+section UnifTight
+
+variable {f g : ι → α → β} {p : ℝ≥0∞}
+
+
+/-- A sequence of functions `f` is uniformly tight in `L^p` if for all `ε > 0`, there
+exists some measurable set `s` with finite measure such that the Lp-norm of
+`f i` restricted to `sᶜ` is smaller than `ε` for all `i`. -/
+def UnifTight {_ : MeasurableSpace α} (f : ι → α → β) (p : ℝ≥0∞) (μ : Measure α) : Prop :=
+  ∀ ⦃ε : ℝ≥0∞⦄, 0 < ε → ∃ s : Set α, μ s ≠ ∞ ∧ ∀ i, snorm (sᶜ.indicator (f i)) p μ ≤ ε
+
+end UnifTight
+
+-- main results and proofs to be filled in
```

Comment:
Sorry, I guess I really didn't understand what you meant by self-contained. Can you clarify whether the this PR should reproduce the contents of `section UnifTight` in the original PR, or just the smaller `namespace UnifTight`?

<probably uninformed opinion>It seems to me that we are getting dangerously close to creating pull requests for the sake of creating pull requests. Whatever logical structure can be created at the level of the hierarchy of PRs, it seems to me, is already present in the original UnifTight.lean source file, at the level of code sections and namespaces.</opinion>

Reply:
The problem with #9163 as it stands is that its PR title does not in any way indicate that it's introducing a definition of uniform tightness. So either:
* You make the title of #9163 be about uniform tightness
* You split off the definition of uniform tightness (and anything that is more relevant to uniform tightness than the extension of Vitali convergence) to a separate PR (this one).

Depending on how you title the PR, having everything in one PR implies either
* that uniform tightness is mathematically used exclusively as an input to the more general version of Vitali convergence
* that the more general version of Vitali convergence is API to the definition of uniform tightness

I definitely disagree with the first, and probably also disagrees with the second.

Reply:
Pragmatically, I would expect the current PR to contain as much as possible from `MeasureTheory.Function.UnifTight` in #9163 that doesn't require changes to other files. Hopefully that represents the bulk of the file, meaning that we can get most of it merged quickly while other dependencies of #9163 are getting reviewed. Then #9163 will contain not much more than the extension of Vitali convergence.

Reply:
OK, thanks. That clarifies what you were after. I will try to do the split later today.

> Depending on how you title the PR, having everything in one PR implies either
> 
>     * that uniform tightness is mathematically used exclusively as an input to the more general version of Vitali convergence
> 
>     * that the more general version of Vitali convergence is API to the definition of uniform tightness
> 
> I definitely disagree with the first, and probably also disagrees with the second.

While you are certainly right at the level of mathematics, arguably, at the level of Mathlib *right now* either one of these possibilities is as good as true. Perhaps I'm being naive here, but it's hard for me to see how the PR hierarchy (which AFAIK disappears into the aether as soon as all the PRs are merged) is relevant to the variation in the domain of use of tightness vs Vitali in Mathlib *in the future*. If there's some point of principle here about structuring PRs, perhaps it could be explained better to beginners. //opinion

Reply:
Well, you definitely want the very few people who care about uniform tightness or Vitali convergence to comment on your PRs and having a PR title that advertises your PR as relevant is a great first step. Note that, even if **you** think topics A and B are very closed, there will be people who care about A but not B, or B but not A. I can't count anymore the number of PRs which I should have reviewed but missed because they had a misleading title/a title about other changes that were lumped with the ones I cared about.

On a more long-termy note, commits are part of the history of Mathlib forever, so it's definitely worth spending *some* time thinking about how to present things to future git archeologists. Eg early PRs like https://github.com/leanprover-community/mathlib/pull/327 are really hard to understand and contain much less documentation than they should have. Splitting PRs really helps here.

Reply:
> Pragmatically, I would expect the current PR to contain as much as possible from `MeasureTheory.Function.UnifTight` in #9163 that doesn't require changes to other files. Hopefully that represents the bulk of the file, meaning that we can get most of it merged quickly while other dependencies of #9163 are getting reviewed. Then #9163 will contain not much more than the extension of Vitali convergence.

Alright, probably this version of the PR is closer to what you were expecting. It still spans multiple files, due to the rearrangements that Yury had done earlier.

If you find some code/comment formatting changes that you would like to make, just go ahead and suggest them. I'm quite indifferent about them and I doubt that I would find all of them myself.